pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -460,3 +460,68 @@ load 6m
 
 eval range from 0 to 96m step 6m round(series, scalar(toNearest))
   {a="b"}  -5   2.7  0.5   20    10    3.25    0         1000.01 0          -1000000  NaN NaN NaN NaN _ _ 10
+
+clear
+
+# Testing resets
+load 1m
+  simple_metric_all_same_no_reset{num="0"} 0 0 0 0 0 0 0 0
```

Comment:
[nit] It might be clearer to change these series to use labels like these:

```
metric{case="all the same, no resets, value 0"}
metric{case="all the same, no resets, value 3"}
metric{case="all floats, no resets"}
metric{case="all floats, some resets"}
...
```

This would make the cases clearer given we can use full English phrases. Then the `eval`s below can be collapsed into one `eval range from 0 to X step Y resets(metric)` as well.

Reply:
Refactored the test in 647feb3c9f2e3a97d8c6ce2ecb523b8248052b58